src/browserlib/extract-dfns.js

```diff
@@ -19,72 +19,451 @@
+    }
+    let iface = idlInterfaces.find(i => i.name.toLowerCase() === id || i.name.toLowerCase() === `html${id}element`);
+    if (iface) {
+      return iface.name;
+    }
+  }
+
+  function fromIdToTypeAndFor(containerid, id) {
+    // deals with exceptions to how containerid / id are expected to be parsed
+    if (id) {
+    [containerid, id] = {
+      "history-scroll": ["history", "scrollrestoration"],
+      // overloads
+      "document-open" : ["document", "open"],
+      "dedicatedworkerglobalscope-postmessage": ["dedicatedworkerglobalscope", "postmessage"],
+      "messageport-postmessage": ["messageport", "postmessage"],
+      "window-postmessage": ["window", "postmessage"],
+      "worker-postmessage": ["worker", "postmessage"],
+      "context-2d-settransform": ["context-2d", "settransform"]
+    }[containerid] || [containerid, id];
+    }
+
+
+    const exceptions = {
+      "worker-navigator": "WorkerGlobalScope",
+      "navigator-canplaytype": "HTMLMediaElement",
+      "media-getsvgdocument": "HTMLIFrameElement,HTMLEmbedElement,HTMLObjectElement",
+      "fe-autofocus": "HTMLOrSVGElement"
+    };
+
+    let interfaces = [];
+    const mixins = {
+      "context-2d": "CanvasRenderingContext2D",
+      "navigator": "Navigator"
+    };
+    const fullId = containerid + "-" + id;
+    if (exceptions[fullId]|| fromIdToIdl(containerid)) {
+      let names = (exceptions[fullId] ? exceptions[fullId] : fromIdToIdl(containerid)).split(",");
+      interfaces = idlInterfaces.filter(i => names.includes(i.name));
+    }
+    if (Object.keys(mixins).includes(containerid)) {
+    // some container id are split across several mixins, lets find out which
+      const candidateInterfaceNames = [mixins[containerid]].concat(idlTree.filter(inc => inc.type === "includes" && inc.target === mixins[containerid]).map(inc => inc.includes));
+      interfaces =  candidateInterfaceNames.map(name => idlInterfaces.filter(iface => iface.name === name)).flat().filter(iface => iface && iface.members && iface.members.find(member => member.name.toLowerCase() === id));
+    }
+
+    if (interfaces.length) {
+      let type = "attribute";
+      let relevantInterfaces = interfaces;
+      if (id) {
+        type = "dfn";
+        // dom-head-profile, intentionally omitted from IDL fragment
+        if (id === "profile" && containerid === "head") {
+          return {type: "attribute", _for:"HTMLHeadElement"};
+        }
+        relevantInterfaces = interfaces.filter(iface => iface.members.find(member => member.name && member.name.toLowerCase() === id));
+        if (relevantInterfaces.length) {
+          let idlTerm = relevantInterfaces[0].members.find(member => member.name && member.name.toLowerCase() === id);
+          type = idlTerm.type === "operation" ? "method" : idlTerm.type;
+        }
+      }
+      return {type, _for: [... new Set(relevantInterfaces.map(iface => iface.name))].join(",")};
+    }
+
+    const enumName = id => { switch(id) {
```

Comment:
Could you move the `switch` to a new line, and move closing `}` back to "normal" place?